### PR TITLE
Feat: Add deeplinking into entity pages

### PR DESCRIPTION
**Pull request by danwom**
### Summary of Changes
- Add support to deeplink into specific tabs on the various entity pages
- Add support to pre-select a column.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/js/pages/DashboardPage/constants.ts [hunk 1]
```diff
@@ -8,3 +8,13 @@ export const EDIT_DESC_TEXT = 'Click to edit description in';
 
 export const LAST_RUN_SUCCEEDED = 'succeeded';
 export const STATUS_TEXT = 'Status:';
+
+export const TABLES_TAB_TITLE = 'Tables';
+export const CHARTS_TAB_TITLE = 'Charts';
+export const QUERIES_TAB_TITLE = 'Queries';
+
+export enum DASHBOARD_TAB {
+  TABLE = 'table',
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/pages/DashboardPage/constants.ts:17]: We changed from 'tables' to 'table', was this something we wanted?

**Review by Golodhros — COMMENTED** [review]
Looks good, only one comment

File: frontend/amundsen_application/static/js/pages/DashboardPage/constants.ts (hunk 1 above)
Comment (danwom) [reply frontend/amundsen_application/static/js/pages/DashboardPage/constants.ts]: Changed back to `tables`. Should be consistent across all tabs showing tables now. Good catch.

File: frontend/amundsen_application/static/js/pages/ProfilePage/index.tsx [hunk 2]
```diff
@@ -157,8 +162,6 @@ export class ProfilePage extends React.Component<
     );
   };
 
-  generateTabKey = (resource: ResourceType) => `tab:${resource}`;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/pages/ProfilePage/index.tsx:160]: Are we sure this `tab:` part wasn't used by anything?

**Review by Golodhros — APPROVED** [review]
LGTM once the linting errors are fixed!

File: frontend/amundsen_application/static/js/pages/ProfilePage/index.tsx (hunk 2 above)
Comment (danwom) [reply frontend/amundsen_application/static/js/pages/ProfilePage/index.tsx:160]: There was a test for it but I deleted this. `tab:${resource}` shows up in the URL as `?tab=tab%3Atable` which isn't pretty.

Comment (Golodhros) [reply frontend/amundsen_application/static/js/pages/ProfilePage/index.tsx:160]: Cool, just making sure it wasn't important

**Pull request merged**